src/peer.rs

```diff
@@ -128,12 +128,20 @@ pub(crate) fn new_peer<T: HandshakeHandler>(
         // SPAWN WRITING THREAD
         //TODO: Bound
         let mut write_endpoint = endpoint.clone();
+        let write_active_connections = active_connections.clone();
+        let write_peer_id = peer_id.clone();
         // https://github.com/crossbeam-rs/crossbeam/issues/288
         let write_thread_handle = std::thread::spawn(move || loop {
             match high_write_rx.try_recv() {
                 Ok(data) => {
                     println!("writer thread: high priority message received");
-                    write_endpoint.send(&data).unwrap();
+                    if write_endpoint.send(&data).is_err() {
+                        write_active_connections
+                            .write()
+                            .connections
+                            .remove(&write_peer_id)
+                            .expect("Unable to remove peer id");
```

Comment:
I think we should stop the peer thread as the channel to communicate with him doesn't work anymore. Same for the other lines added above.